app/models/course_team.rb

```diff
@@ -65,9 +65,20 @@ def self.export_fields(options)
   def add_member(user, _id = nil)
     raise "The user \"#{user.name}\" is already a member of the team, \"#{name}\"" if user?(user)
 
-    t_user = TeamsUser.create(user_id: user.id, team_id: id)
+    add_participant(parent_id, user)
+    participant = CourseParticipant.find_by(parent_id: parent_id, user_id: user.id)
+    t_user = TeamsUser.create(participant_id: participant.id, team_id: id)
+    parent = TeamNode.find_by(node_object_id: id)
+    TeamUserNode.create(parent_id: parent.id, node_object_id: t_user.id)
```

Comment:
Again, why are we using user_id?

Reply:
We're finding participant id based on the id from Users table and assignment id, the creating a new TeamsUser. To display in the team in tree structure, TeamUserNode is created with **id** from teams_users table